src/reducers/utxo_by_address.rs

```diff
@@ -23,6 +24,16 @@ impl Reducer {
         tx_idx: usize,
         output: &mut super::OutputPort,
     ) -> Result<(), gasket::error::Error> {
+        if self.config.filter.is_some() {
```

Comment:
I think you should explore Rust's `if let` syntax, it will probably result in something more compact (won't change logic, just more ergonomic).

Reply:
Implemented. Better like that? :)